src/engine/renderer/tr_shader.cpp

```diff
@@ -5992,209 +5990,154 @@ A second parameter will cause it to print in sorted order
 */
 void R_ShaderList_f()
 {
-	int      i;
-	int      count;
-	shader_t *shader;
-	const char *s = nullptr;
-
-	Log::Notice("-----------------------" );
-
-	if ( ri.Cmd_Argc() > 1 )
-	{
-		s = ri.Cmd_Argv( 1 );
-	}
-
-	count = 0;
-
-	for ( i = 0; i < tr.numShaders; i++ )
-	{
-		if ( ri.Cmd_Argc() > 2 )
-		{
-			shader = tr.sortedShaders[ i ];
-		}
-		else
-		{
-			shader = tr.shaders[ i ];
-		}
-
-		if ( s && !Com_Filter( s, shader->name, false ) )
+	const char *prefix = ri.Cmd_Argc() > 1 ? ri.Cmd_Argv( 1 ) : nullptr;
+
+	using shaderTypeStringMap = std::unordered_map<shaderType_t, std::string>;
```

Comment:
All of this processing sounds like you could store these in static/global variables and use std::once to do it once instead of every call.

WDYT?

Reply:
I googled `std::once` and didn't get any results. This is a rarely used debug command so fine as it is IMO

Reply:
I meant this; https://en.cppreference.com/w/cpp/thread/call_once. But yeah, purely an optional change.

Reply:
Do you mean caching the result of those values across loop run / command call? Well not only this is a debug command, but it has no effect in rendering performance, so I would even don't mind doing slow things there if that made code shorter…